lib/server/ServerRenderingWorkflow.js

```diff
@@ -12,90 +12,86 @@ var validateView = require("../util/validateView");
 
 var ServerRenderingWorkflow = {
 
-    createHandlerFrom: function(originalHandler) {
-        return function(params, expressRequest, environmentConfig, clientAppRequirePath) {
-            var serverRequest = ServerRequest.from(expressRequest, environmentConfig);
-            var serverResponse = ServerResponse.create();
-
-            var routeArguments = params;
-            var router = this;
-            var LayoutForRoute = router.layout;
-            var errorViewMapping = router.errorViewMapping;
-            var layout = new LayoutForRoute();
-            var layoutDelegate = LayoutDelegate.from(layout, LayoutForRoute);
-
-            layout.setEnvironmentConfig(environmentConfig);
-
-            var fetchLayoutData = Promise.method(function() {
-                return layout.fetchData();
-            });
-
-            // If handler returns a view, this function returns a promise of a view
-            // If handler returns a promise of a view, this function returns a promise of a view
-            // and absorbs the state of the returned promise
-            var chooseViewForRoute = Promise.method(function() {
-                var allParams = addExtraParamsTo(routeArguments,
-                    layoutDelegate,
-                    serverRequest,
-                    serverResponse
-                );
-
-                return originalHandler.apply(router, allParams);
-            });
-
-            function renderLayout(view) {
-                layout.render();
-
-                return view;
-            }
+    execute: function(router, originalHandler, params, expressRequest, environmentConfig, clientAppRequirePath) {
```

Comment:
the only things changed in this file:

- function signature: router passed in
- function returns promise of client rendering rather than a function